Mixpanel/SelectorEvaluator.swift

```diff
@@ -0,0 +1,604 @@
+//
+//  Selector.swift
+//  Mixpanel
+//
+//  Created by Madhu Palani on 3/1/19.
+//  Copyright © 2019 Mixpanel. All rights reserved.
+//
+
+import Foundation
+
+extension DateFormatter {
+    static func formatterForJSONDate() -> DateFormatter {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss"
+        formatter.locale = Locale(identifier: "en_US_POSIX")
+        return formatter
+    }
+}
+
+class SelectorEvaluator {
+    // Key words
+    static let OPERATOR_KEY = "operator"
+    static let CHILDREN_KEY = "children"
+    static let PROPERTY_KEY = "property"
+    static let VALUE_KEY = "value"
+    static let EVENT_KEY = "event"
+    static let LITERAL_KEY = "literal"
+    static let WINDOW_KEY = "window"
+    static let UNIT_KEY = "unit"
+    static let HOUR_KEY = "hour"
+    static let DAY_KEY = "day"
+    static let WEEK_KEY = "week"
+    static let MONTH_KEY = "month"
+    // Typecast operators
+    static let BOOLEAN_OPERATOR = "boolean"
+    static let DATETIME_OPERATOR = "datetime"
+    static let LIST_OPERATOR = "list"
+    static let NUMBER_OPERATOR = "number"
+    static let STRING_OPERATOR = "string"
+    // Binary operators
+    static let AND_OPERATOR = "and"
+    static let OR_OPERATOR = "or"
+    static let IN_OPERATOR = "in"
+    static let NOT_IN_OPERATOR = "not in"
+    static let PLUS_OPERATOR = "+"
+    static let MINUS_OPERATOR = "-"
+    static let MUL_OPERATOR = "*"
+    static let DIV_OPERATOR = "/"
+    static let MOD_OPERATOR = "%"
+    static let EQUALS_OPERATOR = "=="
+    static let NOT_EQUALS_OPERATOR = "!="
+    static let GREATER_THAN_OPERATOR = ">"
+    static let GREATER_THAN_EQUAL_OPERATOR = ">="
+    static let LESS_THAN_OPERATOR = "<"
+    static let LESS_THAN_EQUAL_OPERATOR = "<="
+    // Unary operators
+    static let NOT_OPERATOR = "not"
+    static let DEFINED_OPERATOR = "defined"
+    static let NOT_DEFINED_OPERATOR = "not defined"
+    // Special words
+    static let NOW_LITERAL = "now"
+    
+    let selector: [String:Any]?
+    
+    init?(JSONObject: [String: Any]?) {
```

Comment:
nit: 
```suggestion
    init?(jsonObject: [String: Any]?) {
```
The same as above

Reply:
can SelectorEvaluator even be a stateless class so you don't even need `init`?